modules/aks_node_pool/main.tf

```diff
@@ -9,12 +9,21 @@ resource "azurerm_kubernetes_cluster_node_pool" "autoscale_node_pool" {
   proximity_placement_group_id = var.proximity_placement_group_id == "" ? null : var.proximity_placement_group_id
   vm_size                      = var.machine_type
   os_disk_size_gb              = var.os_disk_size
+  # TODO: enable after azurerm v2.37.0
```

Comment:
Azurerm 2.38.0 has been out for about a month now. Is the need for `os_disk_type` tied to this if so lets make the update.

Reply:
limiting the scope here to these changes not knowing all the changes that v2.38.0 or the latest v2.40.0 would require.